src/docs/user/ProActiveUserGuide.adoc

```diff
@@ -1763,6 +1710,234 @@ println outFile.text
+
+In some cases, *static task dependencies* are not enough to synchronize tasks execution.
+
+This is the case when a task depends on the state of another task executed by a different branch of a workflow, a sibling task or even another job/workflow.
+
+For example, let's suppose workflow _A_ contains a single task _A_1_ starting an apache server. Task _A_1_ starts the server and remains alive while the server is up.
+When workflow _A_ is killed, task _A_1_ stops the apache server.
+
+A task _B_1_ of another workflow _B_ wants to interact with this server.
+
+*Problem:* task _B_1_ does not know if the apache server is up and how to contact it.
+
+Using *Task Synchonization API*, task _A_1_ can advertise the server URL to the Synchronization Service.
+
+Task _B_1_ can use Task Synchonization API to wait until the apache server is advertised, it can also perform the necessary operations and advertise task _A_1_ that the operation is complete and the apache server can be terminated.
+
+As we can see, Task Synchronization API can be used for two-ways synchronization.
+
+==== API Usage
+
+*Task Synchronization API* is a *Java API* defined by the following https://www.activeeon.com/public_content/documentation/javadoc/dev/org/ow2/proactive/scheduler/synchronization/Synchronization.html[API documentation].
+
+The Synchronization API is connected to a centralized *Synchronization Service*. This service ensures that each request is executed atomically, and that changes are persisted on disk.
+
+The *Synchronization Service* works as a _Key-Value Store_ organized in *channels*. Each *channel* acts as a private _Key-Value Store_ or _Hash Map_.
+
+This means that each _put_ operation is identified by a 3-tuple: _<Channel_identifier, Key_binding, Value>_.
+
+The Synchronization API is available inside ProActive Tasks through the `synchronizationapi` <<_glossary_script_bindings,script binding>>.
+
+Users of the Synchronization API are responsible to create *channels* as fit their needs. For example, a channel which is used inside a specific job should be created inside some *initialization task* at the beginning of this job, and deleted in some *cleaning task* at the end.
+When channels are created, the user can decide if the channel should be persisted on disk or kept in memory only.
+
+The Synchronization API is similar to the https://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Java 8 Map interface], with the following differences:
+
+ * Each operation must take a _channel identifier_ parameter. Some methods like _createChannel_ allows to manipulate channels.
+
+ * The Synchronization API supports https://docs.oracle.com/javase/tutorial/java/javaOO/lambdaexpressions.html[Java 8 lambdas]. All methods present in the https://docs.oracle.com/javase/8/docs/api/java/util/Map.html[Java 8 Map interface] are implemented, but lambdas must be passed as string arguments defining http://groovy-lang.org/closures.html[Groovy Closures].
```

Comment:
This sentence should be about lambdas only (we already explained that it is Java Map-based before). However, it looks like the user will be able to pass Java lambdas, whereas he actually have to write Groovy lambdas. Maybe it is better to only present it this way?

Reply:
ok